pkg/logs/input/file/scanner.go

```diff
@@ -25,7 +25,8 @@ const scanPeriod = 10 * time.Second
 // or update the old ones if needed
 type Scanner struct {
 	pipelineProvider    pipeline.Provider
-	sources             chan *config.LogSource
+	addedSources        chan *config.LogSource
+	removedSources      chan *config.LogSource
 	activeSources       []*config.LogSource
```

Comment:
For later, could be worth exporting this as a metric (maybe something that is local to this file instead of using the global logs metrics). This would allow us to monitor leaks after releases.